### PR TITLE
LOOKDEVX-2128 - Allow specializing the topo neutral graph generator

### PR DESCRIPTION
**Pull request by JGamache-autodesk**
- Also exports the color management preferences cache.

### CODE REVIEW

File: lib/mayaUsd/render/vp2RenderDelegate/colorManagementPreferences.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    ColorManagementPreferences();
+    static const ColorManagementPreferences& Get();
+    static ColorManagementPreferences&       InternalGet();
+
+    bool                     _dirty = true;
+    bool                     _active = false;
+    MString                  _renderingSpaceName;
+    MString                  _sRGBName;
+    std::vector<MCallbackId> _mayaColorManagementCallbackIds;
+    MCallbackId              _mayaExitingCB { 0 };
```
Comment (feldstj) [inline lib/mayaUsd/render/vp2RenderDelegate/colorManagementPreferences.h:77]: Can add this callback to the color management callback vector.

**Review by feldstj — APPROVED** [review]
LGTM, one minor comment.

**Review by feldstj — APPROVED** [review]

**Comment by JGamache-autodesk** [comment]
@seando-adsk Ready for merge.

**Comment by JGamache-autodesk** [comment]
@seando-adsk  Still ready for merge. Last commit fully reverts previous one, which will be submitted separately. So we are still in a fully reviewed state.

**Review by feldstj — APPROVED** [review]

File: lib/mayaUsd/render/vp2RenderDelegate/material.cpp [hunk 2]
```diff
@@ -3027,8 +2910,15 @@ MHWRender::MShaderInstance* HdVP2Material::CompiledNetwork::_CreateMaterialXShad
             return shaderInstance;
         }
 
+        // Enable changing texcoord to geompropvalue
+        const auto prevUVSetName = mx::OgsXmlGenerator::getPrimaryUVSetName();
+        mx::OgsXmlGenerator::setPrimaryUVSetName(_GetMaterialXData()._mainUvSetName);
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2RenderDelegate/material.cpp:2915]: Fixes issue see in demo this morning.

**Review by feldstj — APPROVED** [review]

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**